src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+
+### Exporting Configuration
+
+The weights and biases will be output into a yaml file like the one found at [model.yaml on the Visual Mesh repository](https://github.com/Fastcode/VisualMesh/blob/master/example/model.yaml) or [VisualMesh.yaml on the NUbots repository](https://github.com/NUbots/NUbots/blob/master/module/vision/VisualMesh/data/config/VisualMesh.yaml).
+
+Replace the NUbots repository `VisualMesh.yaml` configuration with the configuration you just obtained from training the Visual Mesh. Ensure you create a new branch so you can make a PR and get any changes merged in. For further information about Git, view the [Git Guide](/guides/general/git).
+
+## Camera Calibration
+
+If the cameras are not calibrated correctly, the vision system will not work well. Information on calibration is available on [the input page](/system/subsystems/input).
+
+An automatic camera calibration tool is available to use in the NUbots repository that can be used to calibrate the cameras. A guide on how to calibrate the cameras will be available on NUbook soon.
+
+## Testing
+
+Now that the Visual Mesh has been updated in the NUbots repository, you should test it before merging.
+
+Build the code, ensuring `VisualMesh.role` is on in `./b configure -i`. Go to the [Getting Started page](/guides/main/getting-started) to find out more on how to build the code and install onto a robot. Ensure the new configuration file has been installed. Check out the [Build System page](/system/foundations/build-system) to find out more about options when installing onto the robot.
+
+When your new Visual Mesh is installed onto the robot, connect to the robot and ensure NUsight is on.
+
+```sh
+nano config/NUsight.yaml
+```
+
+Turn `vision object` and `compressed images` on. Run NUsight using `yarn prod` and navigate to the NUsight page in your browser. More on NUsight can be found on [the NUsight page](/system/tools/nusight).
+
+Run the visualmesh role
+
+```sh
+./visualmesh
+```
+
+Wait for the cameras to load and then watch the Vision tab in NUsight. To determine if the output is correct, consult the [vision page](/system/subsystems/vision) for images of the expected output.
+
+If you would like to see the visual mesh output in NUsight, you will need to log the data and run it back in NUsight using DataPlayback, since the data is too large to send over a network. Add into the `visualmesh.role` file `support::logging::DataLogging`. Build to the robot and rerun the visual mesh binary with `./visualmesh`. An NBS file will be created on the robot in `logs/visualmesh/`. Copy this across the network to the computer you built on using `scp <robot_address>:logs/visualmesh/<file_name> ./recordings` on the computer in the NUbots folder. Read [the Logging page](/system/subsystems/logging) to find out how to play an NBS file back.
```

Comment:
Should we mention that the expect framerate of the visualmesh footage viewed through NUsight is terrible? One might think that something was wrong with what they were doing, when it's really just a question of the bandwidth iirc.

Reply:
That's pretty much what the paragraph above says, you can't run it in realtime on NUsight because it's too large to send over a network so log it and play it back.

Reply:
Another alternative to this is to use the nbs playback functionality in nusight

Reply:
There is standalone NUsight functionality to play back an NBS file?

Reply:
NUsight NBS playback seems to not work great, error'd after a few loops when I tried it. Until it's fixed or someone proves it works (and I missed something) I won't document it here 😞